### PR TITLE
Dịch "fm_vn" - Phần 2

### LINKED ISSUE #2714: Dịch "fm_vn" - Phần 2
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_recommender-systems/fm_vn.md#L68-L233

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by trGiang99**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Điền số issue sau dấu `#` (không có dấu cách): Close #2714

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy thảo luận thêm ở thẻ Discussions của dự án.

* [x] Format của bản gốc có được giữ nguyên không?
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn. Bạn đã kiểm tra rằng mỗi đoạn văn đã nằm một dòng?

* [x] Điền tên đầy đủ của bạn vào mục **Những người thực hiện** ở cuối tập tin.

## Lưu ý cho người phản biện (Reviewer)

1. Chỉ review khi Pull Request (PR) này đã sẵn sàng cho việc review. Một PR sẵn sàng khi PR đó không có label `status: wip`.
2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

### CODE REVIEW

File: chapter_recommender-systems/fm_vn.md [hunk 1]
```diff
@@ -99,15 +101,17 @@ With this reformulation, the model complexity are decreased greatly.
 Moreover, for sparse features, only non-zero elements needs to be computed so that the overall complexity is linear to the number of non-zero features.
 -->
 
-*dịch đoạn phía trên*
+Với biến đổi này, độ phức tạp của mô hình giảm đi đáng kể.
+Hơn nữa, với đặc trưng thưa, chỉ các phần tử khác không cần phải tính toán nên độ phức tạp toàn phần quan hệ tuyến tính với số đặc trưng khác không.
```
Comment (rootonchair) [inline chapter_recommender-systems/fm_vn.md:105]:
```suggestion
Hơn nữa, với đặc trưng thưa, chỉ các phần tử khác 0 cần phải tính toán nên độ phức tạp toàn phần có quan hệ tuyến tính với số đặc trưng khác 0.
```

File: chapter_recommender-systems/fm_vn.md [hunk 2]
```diff
@@ -99,15 +101,17 @@ With this reformulation, the model complexity are decreased greatly.
...
 
 
 <!--
 To learn the FM model, we can use the MSE loss for regression task, the cross entropy loss for classification tasks, and the BPR loss for ranking task.
 Standard optimizers such as SGD and Adam are viable for optimization.
 -->
 
-*dịch đoạn phía trên*
+Để học mô hình FM, ta có thể sử dụng mất mát MSE cho tác vụ hồi quy, mất mát entropy chéo với tác vụ phân loại, và mất mát BPR với tác vụ xếp hạng.
+Các bộ tối ưu chuẩn như SGD và Adam đều khả thi để tối ưu.
```
Comment (rootonchair) [inline chapter_recommender-systems/fm_vn.md:114]:
```suggestion
Các bộ tối ưu chuẩn như SGD và Adam đều khả thi cho việc tối ưu.
```

File: chapter_recommender-systems/fm_vn.md [hunk 3]
```diff
@@ -228,7 +236,8 @@ d2l.train_ch13(net, train_iter, test_iter, loss, trainer, num_epochs, devices)
 * Vary the embedding size to check its impact on performance, can you observe a similar pattern as that of matrix factorization?
 -->
 
-*dịch đoạn phía trên*
+* Bạn có thể kiểm tra FM trên một tập dữ liệu khác ví dụ như tập dữ liệu Avazu, MovieLens, and Criteo không?
+* Thay đổi kích thước embedding để kiểm tra ảnh hưởng của nó lên hiệu năng, liệu bạn có thể quan sát thấy kiểu mẫu tương tự như vậy với phân rã ma trận?
```
Comment (rootonchair) [inline chapter_recommender-systems/fm_vn.md:240]:
```suggestion
* Thay đổi kích thước embedding để kiểm tra ảnh hưởng của nó lên hiệu năng, liệu bạn có thể quan sát thấy khuôn mẫu tương tự đối với phân rã ma trận?
```

**Review by rootonchair — COMMENTED** [review]
Thanks

File: chapter_recommender-systems/fm_vn.md [hunk 4]
```diff
@@ -251,6 +260,6 @@ Tên đầy đủ của các reviewer có thể được tìm thấy tại https
 * 
 
 <!-- Phần 2 -->
-* 
+* Đỗ Trường Giang
```
Comment (trGiang99) [inline chapter_recommender-systems/fm_vn.md:263]:
```suggestion
* Đỗ Trường Giang
* Phạm Hồng Vinh
```

File: chapter_recommender-systems/fm_vn.md [hunk 5]
```diff
@@ -133,7 +137,9 @@ It is clear to see that FM consists a linear regression block and an efficient f
 We apply a sigmoid function over the final score since we treat the CTR prediction as a classification task.
 -->
 
-*dịch đoạn phía trên*
+Đoạn mã sau đây lập trình cho mô hình máy phân rã ma trận.
```
Comment (cuongvng) [inline chapter_recommender-systems/fm_vn.md:140]:
```suggestion
Đoạn mã sau đây lập trình mô hình máy phân rã ma trận.
```

File: chapter_recommender-systems/fm_vn.md [hunk 6]
```diff
@@ -189,15 +195,16 @@ test_iter = gluon.data.DataLoader(
...
+## Huấn luyện mô hình
 
 
 <!--
 Afterwards, we train the model. The learning rate is set to 0.01 and the embedding size is set to 20 by default.
 The `Adam` optimizer and the `SigmoidBinaryCrossEntropyLoss` loss are used for model training.
 -->
 
-*dịch đoạn phía trên*
+Cuối cùng, ta tiến hành huấn luyện mô hình. Tốc độ học được đặt bằng 0.01 và kích thước embedding mặc định bằng 20.
```
Comment (cuongvng) [inline chapter_recommender-systems/fm_vn.md:206]:
```suggestion
Cuối cùng, ta tiến hành huấn luyện mô hình. Tốc độ học được đặt bằng 0.02 và kích thước embedding mặc định bằng 20.
```
dưới code là vậy

File: chapter_recommender-systems/fm_vn.md [hunk 7]
```diff
@@ -218,7 +225,8 @@ d2l.train_ch13(net, train_iter, test_iter, loss, trainer, num_epochs, devices)
 * Feature interaction/crossing is important for prediction tasks and the 2-way interaction can be efficiently modeled with FM.
 -->
 
-*dịch đoạn phía trên*
+* FM là một framework khái quát có thể áp dụng cho nhiều tác vụ khác nhau như hồi quy, phân loại hay xếp hạng.
+* Tương tác/giao đặc trưng (*feature interaction/crossing*) rất quan trọng trong tác vụ dự đoán và tương tác hai chiều có thể được mô hình hoá một cách hiệu quả với FM.
```
Comment (cuongvng) [inline chapter_recommender-systems/fm_vn.md:229]:
```suggestion
* Tương tác/tương giao đặc trưng (*feature interaction/crossing*) rất quan trọng trong tác vụ dự đoán và tương tác hai chiều có thể được mô hình hoá một cách hiệu quả với FM.
```

**Review by cuongvng — APPROVED** [review]

File: chapter_recommender-systems/fm_vn.md [hunk 8]
```diff
@@ -251,6 +260,7 @@ Tên đầy đủ của các reviewer có thể được tìm thấy tại https
 * 
 
 <!-- Phần 2 -->
-* 
+* Đỗ Trường Giang
+* Phạm Hồng Vinh
```
Comment (trGiang99) [inline chapter_recommender-systems/fm_vn.md:264]:
```suggestion
* Phạm Hồng Vinh
* Nguyễn Văn Cường
```

**Review by duythanhvn — APPROVED** [review]
